cypress/integration/articleFlows/commentOnArticle.spec.js

```diff
@@ -0,0 +1,132 @@
+  });
+
+  it('should accept entered comment text without user mention if no autocomplete suggestions', () => {
+    cy.findByLabelText('Add a comment to the discussion').as('commentBox');
+    startAutocompleteInTextArea('@commentBox');
+
+    cy.findByLabelText('mention user').as('comboboxHiddenInput');
+    cy.get('@comboboxHiddenInput').should('have.focus');
+
+    cy.get('@comboboxHiddenInput').type('user');
+
+    cy.findByText('No results found').should('exist');
+    cy.get('@comboboxHiddenInput').type(' ');
+    cy.findByText('No results found').should('not.exist');
+
+    cy.findByRole('button', { name: /Submit/ }).click();
+    cy.findByRole('link', { name: /@user/ }).should('not.exist');
+    cy.findByDisplayValue('Some text @user').should('exist');
+  });
+
+  it('should update mention autocomplete suggestions on text delete', () => {
+    cy.findByLabelText('Add a comment to the discussion').as('commentBox');
+    startAutocompleteInTextArea('@commentBox');
+
+    cy.findByLabelText('mention user').as('comboboxHiddenInput');
+    cy.get('@comboboxHiddenInput').should('have.focus');
+
+    cy.get('@comboboxHiddenInput').type('search_user_1');
+    cy.findByText('@search_user_1').should('exist');
+    cy.findByText('@search_user_2').should('not.exist');
+
+    cy.get('@comboboxHiddenInput').type('{backspace}');
+    cy.findByText('@search_user_2').should('exist');
+  });
+
+  it('should close the autocomplete suggestions on Escape press', () => {
+    cy.findByLabelText('Add a comment to the discussion').as('commentBox');
+    startAutocompleteInTextArea('@commentBox');
+
+    cy.findByLabelText('mention user').as('comboboxHiddenInput');
+    cy.get('@comboboxHiddenInput').should('have.focus');
+
+    cy.get('@comboboxHiddenInput').type('search');
+    cy.findByText('@search_user_1').should('be.visible');
+
+    cy.get('@comboboxHiddenInput').type('{Esc}');
+    cy.findByText('@search_user_1').should('not.be.visible');
+  });
+
+  it('should reply to a comment with user mention autocomplete', () => {
+    cy.findByLabelText('Add a comment to the discussion').as('commentBox');
+    cy.get('@commentBox').type('first comment');
+    cy.findByRole('button', { name: /Submit/ }).click();
+
+    cy.findByRole('link', { name: /Reply/ }).click();
+    cy.findByLabelText('Reply to a comment...').as('replyCommentBox');
+
+    startAutocompleteInTextArea('@replyCommentBox');
+
+    cy.findByLabelText('mention user').as('comboboxHiddenInput');
+    cy.get('@comboboxHiddenInput').should('have.focus');
+    cy.get('@comboboxHiddenInput').type('search_user');
+
+    cy.findByText('@search_user_1').click();
+    cy.findByRole('button', { name: /Submit/ }).click();
+    cy.findByRole('link', { name: /@search_user_1/ }).should('exist');
+  });
+
+  const startAutocompleteInTextArea = (textAreaSelector) => {
+    //   Extra click is a workaround for https://github.com/cypress-io/cypress/issues/5023
```

Comment:
Without the extra click I'm seeing flakiness in the 'focusin' event being triggered, based on whether the cypress browser window is focused or not (e.g. if you start them running and then start clicking around elsewhere on your desktop), as well as running as `cypress:run`

Seems others are encountering this too: https://github.com/cypress-io/cypress/issues/5023